src/k8s-engine/client.ts

```diff
@@ -166,38 +194,43 @@ export class KubeClient extends events.EventEmitter {
     this.forwarder = new k8s.PortForward(this.kubeconfig, true);
     this.shutdown = false;
     this.coreV1API = this.kubeconfig.makeApiClient(k8s.CoreV1Api);
+    this.services = null;
+  }
 
-    // Set up a watch for services
-    // Since the watch API we have _doesn't_ notify us when things have
-    // changed, we'll need to do some trickery and wrap the underlying watcher
-    // with our own code.
-    class WrappedWatch extends k8s.Watch {
-      callback: ()=>void;
-      constructor(kubeconfig: k8s.KubeConfig, callback:()=>void) {
-        super(kubeconfig);
-        this.callback = callback;
-      }
+  async waitForServiceWatcher() {
```

Comment:
Is there any reason this isn't triggered in the constructor?

Reply:
I don't understand why this was resolved? It's still not triggered in the constructor, and there doesn't appear to be comments to explain it…

Reply:
added a comment